### ISSUE TITLE
Open relative link

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura#102]
*On GitLab by @pickfire on Jan 12, 2019, 18:54*

---

Zathura fail to open relative file link. Latex source:

```latex
\href{hello.txt}{Hello}
```

Zathura have an additional `file:/` when opening files. Logs:

```
error: Unknown file type: 'cannot open `/path/to/dir/file:/hello.txt' (No such file or directory)'
```

I believe it supposed to be be `file://path/to/dir/hello.txt` instead.

PDF works in evince.

**Comment by sebastinas** [comment]
mentioned in commit 3f2001e6bc6171ab53e1684187bc3407c119e955

**Comment by sebastinas** [comment]
closed

**Comment by sebastinas** [comment]
*On GitLab by @amosbird on Apr 23, 2019, 05:42*

---

Hmm, I believe this issue isn't resolved yet. `\href{hello.txt}{Hello}` somehow generates a link type `ZATHURA_LINK_GOTO_REMOTE` which is not covered by this commit.

**Comment by sebastinas** [comment]
*On GitLab by @agguser on May 25, 2020, 16:00*

---

This bug is still not fixed! For example, if `/tmp/t.pdf` contains a relative link to `d2/t2.pdf`, Zathura will try to open `/tmp/file:/d2/t2.pdf`, which is wrong. (Tested with `zatura-pdf-mupdf` on Arch Linux.)

By the way, it would be good to show link target (in status bar or as tooltip) when mouse pointer is over a link.

**Comment by sebastinas** [comment]
*On GitLab by @timhae on Aug 14, 2020, 10:50*

---

I also have this isse with (slightly outdated)
```
❯ zathura --version
zathura 0.4.4
girara 0.3.4 (runtime: 0.3.4)
(plugin) djvu (0.2.8) (/nix/store/gri3cspna095zgsllxgrnxksp2af73q6-zathura-djvu-0.2.8/lib/zathura/libdjvu.so)
(plugin) ps (0.2.6) (/nix/store/sklmkl9w7bqqbp3v4lh4b69iinrymmrf-zathura-ps-0.2.6/lib/zathura/libps.so)
(plugin) cb (0.1.8) (/nix/store/pgg7z3094wl6pfrq5c6v1gkyx9023a75-zathura-cb-0.1.8/lib/zathura/libcb.so)
(plugin) pdf-mupdf (0.3.5) (/nix/store/w70swryf5xbbpinjzzlzskk405dqg01b-zathura-pdf-mupdf-0.3.5/lib/zathura/libpdf-mupdf.so)
```

**Comment by sebastinas** [comment]
*On GitLab by @ma27 on Feb 5, 2022, 24:22*

---

mentioned in commit ma27/zathura@05aec96f09bd71c8e069c7d45178c791c899c045

**Comment by sebastinas** [comment]
*On GitLab by @ma27 on Feb 5, 2022, 01:43*

---

mentioned in merge request ma27/zathura!1

**Comment by sebastinas** [comment]
*On GitLab by @ma27 on Feb 5, 2022, 01:44*

---

mentioned in merge request ma27/zathura!2

**Comment by sebastinas** [comment]
*On GitLab by @ma27 on Feb 5, 2022, 01:56*

---

I implemented a potential fix in https://git.pwmt.org/ma27/zathura/-/commit/05aec96f09bd71c8e069c7d45178c791c899c045.

Not sure why, but I can't seem to open a MR against this repo.

cc @sebastinas

**Comment by Ma27** [comment]
@sebastinas why was this closed btw?
The fix should still be relevant?

**Comment by sebastinas** [comment]
The migration script decided to close it. I haven't double checked its results (it's just too many bugs).

**Comment by sebastinas** [comment]
Your fix is now applied as 5e9d542.
